src/explorer/Exploration.cs

```diff
@@ -3,27 +3,31 @@ namespace Explorer
     using System;
     using System.Collections;
     using System.Collections.Generic;
+    using System.Collections.Immutable;
     using System.Linq;
+    using System.Threading;
     using System.Threading.Tasks;
-    using Diffix;
-    using Explorer.Metrics;
+    using Lamar;
+    using static Explorer.ExplorationStatusEnum;
 
     public sealed class Exploration : AbstractExploration, IDisposable
     {
+        private readonly IContainer rootContainer;
+        private readonly ExplorationScopeBuilder scopeBuilder;
+        private readonly CancellationTokenSource cancellationTokenSource;
         private bool disposedValue;
 
-        public Exploration(string dataSource, string table, IEnumerable<ExplorationScope> scopes)
+        public Exploration(
+            IContainer rootContainer,
+            ExplorationScopeBuilder scopeBuilder)
         {
-            DataSource = dataSource;
-            Table = table;
-            ColumnExplorations = scopes.Select(scope => new ColumnExploration(scope)).ToList();
+            this.rootContainer = rootContainer;
+            this.scopeBuilder = scopeBuilder;
+            cancellationTokenSource = new CancellationTokenSource();
```

Comment:
A couple of reasons for this:
- `DataSource` and `Table` are not actually used anywhere in the `Exploration` so it didn't seem to make sense they they are required in the constructor. They were only added so that we could later retrieve them when building the api response, however the api response should not be concern of the `Exploration`.
- We can't pass the context directly since the context is built by the validation step. So instead we pass a context builder object - this is then invoked by the `Exploration` class to build the context for each `ColumnExploration`. 

(See my comments about `ExplorationLauncher` below.)